### PR TITLE
Unit tests patch

### PR DESCRIPTION
**Pull request by hstrychalski**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #105

### CODE REVIEW

File: tests/Data/Formatter/XmlDataResponseFormatterTest.php [hunk 1]
```diff
@@ -39,4 +39,111 @@ public function testFormatterEncoding(): void
...
+        $this->assertSame(
+            "<?xml version=\"1.1\" encoding=\"UTF-8\"?>\n<response>test</response>\n",
+            $result->getBody()->getContents()
+        );
+        $this->assertSame(['application/xml; UTF-8'], $result->getHeader('Content-Type'));
+    }
+
+    public function testFormatterRootTag(): void
+    {
+        $factory = new Psr17Factory();
```
Comment (xepozz) [inline tests/Data/Formatter/XmlDataResponseFormatterTest.php:61]: Please use mock instead

File: tests/Data/Formatter/XmlDataResponseFormatterTest.php [hunk 2]
```diff
@@ -39,4 +39,111 @@ public function testFormatterEncoding(): void
...
+        $dataResponse = new DataResponse('test', 200, '', $factory);
+        $formatter = new XmlDataResponseFormatter();
+        $formatter = $formatter->withRootTag('exampleRootTag');
+        $result = $formatter->format($dataResponse);
```
Comment (xepozz) [inline tests/Data/Formatter/XmlDataResponseFormatterTest.php:65]: You can use fluent interface to configure the `Formatter`:
```php
$result = (new XmlDataResponseFormatter())
                 ->withRootTag('tag')
                 ->format($dataResponse)
```

File: tests/Data/Formatter/XmlDataResponseFormatterTest.php [hunk 3]
```diff
@@ -39,4 +39,111 @@ public function testFormatterEncoding(): void
...
     }
+
+    public function testFormatterVersion(): void
+    {
+        $factory = new Psr17Factory();
+        $dataResponse = new DataResponse('test', 200, '', $factory);
+        $formatter = new XmlDataResponseFormatter();
+        $formatter = $formatter->withVersion('1.1');
+        $result = $formatter->format($dataResponse);
+        $result->getBody()->rewind();
```
Comment (xepozz) [inline tests/Data/Formatter/XmlDataResponseFormatterTest.php:50]: Is it really needed?

**Review by xepozz — CHANGES_REQUESTED** [review]

File: tests/Data/Formatter/XmlDataResponseFormatterTest.php (hunk 3 above)
Comment (samdark) [reply tests/Data/Formatter/XmlDataResponseFormatterTest.php:50]: Yes. See https://github.com/php-fig/http-message/blob/master/src/StreamInterface.php#L137.

File: tests/Data/Formatter/XmlDataResponseFormatterTest.php (hunk 1 above)
Comment (samdark) [reply tests/Data/Formatter/XmlDataResponseFormatterTest.php:61]: Why? We're testing output so we need a working stream to capture it.

**Review by samdark — APPROVED** [review]

**Review by rustamwin — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**